TiUP playground bootstrap failed because of getTS error

## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do?
<!--If possible, provide a recipe for reproducing the error.-->

```
tiup playground v6.1.7 --db 1 --pd 1 --kv 3 --tiflash 0
```

   v6.1.7 makes it extremely easy to reproduce the issue, while the problem cannot be replicated in other versions.

2. What did you expect to see?

Bootstrap a TiDB cluster successfully

3. What did you see instead?

```
[2026/01/24 20:43:35.598 +08:00] [INFO] [store.go:75] ["new store"] [path=tikv://127.0.0.1:2379]
[2026/01/24 20:43:35.598 +08:00] [INFO] [systime_mon.go:26] ["start system time monitor"]
[2026/01/24 20:43:35.598 +08:00] [INFO] [client.go:392] ["[pd] create pd client with endpoints"] [pd-address="[127.0.0.1:2379]"]
[2026/01/24 20:43:38.587 +08:00] [INFO] [base_client.go:275] ["[pd] cannot update member from this address"] [address=http://127.0.0.1:2379] [error="[PD:client:ErrClientGetLeader]get leader from leader address don't exist error"]
[2026/01/24 20:43:39.589 +08:00] [INFO] [base_client.go:350] ["[pd] switch leader"] [new-leader=http://127.0.0.1:2379] [old-leader=]
[2026/01/24 20:43:39.589 +08:00] [INFO] [base_client.go:105] ["[pd] init cluster id"] [cluster-id=0]
[2026/01/24 20:43:39.589 +08:00] [INFO] [client.go:687] ["[pd] tso dispatcher created"] [dc-location=global]
[2026/01/24 20:43:39.590 +08:00] [ERROR] [client.go:847] ["[pd] getTS error"] [dc-location=global] [stream-addr=http://127.0.0.1:2379] [error="[PD:client:ErrClientGetTSO]rpc error: code = FailedPrecondition desc = mismatch cluster id, need 7598907903586024555 but got 0: rpc error: code = FailedPrecondition desc = mismatch cluster id, need 7598907903586024555 but got 0"]
[2026/01/24 20:43:39.590 +08:00] [WARN] [store.go:83] ["new store with retry failed"] [error="rpc error: code = FailedPrecondition desc = mismatch cluster id, need 7598907903586024555 but got 0"]
[2026/01/24 20:43:39.590 +08:00] [FATAL] [terror.go:292] ["unexpected error"] [error="rpc error: code = FailedPrecondition desc = mismatch cluster id, need 7598907903586024555 but got 0"] [stack="github.com/pingcap/tidb/parser/terror.MustNil\n\t/Users/pingcap/workspace/build-common/go/src/github.com/pingcap/tidb/parser/terror/terror.go:292\nmain.createStoreAndDomain\n\t/Users/pingcap/workspace/build-common/go/src/github.com/pingcap/tidb/tidb-server/main.go:294\nmain.main\n\t/Users/pingcap/workspace/build-common/go/src/github.com/pingcap/tidb/tidb-server/main.go:202\nruntime.main\n\t/usr/local/go1.19.10/src/runtime/proc.go:250"] [stack="github.com/pingcap/tidb/parser/terror.MustNil\n\t/Users/pingcap/workspace/build-common/go/src/github.com/pingcap/tidb/parser/terror/terror.go:292\nmain.createStoreAndDomain\n\t/Users/pingcap/workspace/build-common/go/src/github.com/pingcap/tidb/tidb-server/main.go:294\nmain.main\n\t/Users/pingcap/workspace/build-common/go/src/github.com/pingcap/tidb/tidb-server/main.go:202\nruntime.main\n\t/usr/local/go1.19.10/src/runtime/proc.go:250"]
```


[2026/01/24 20:43:39.589 +08:00] [INFO] [base_client.go:105] ["[pd] init cluster id"] [cluster-id=0]

Got wrong cluster-id via initClusterID API

4. What version of TiUP are you using (`tiup --version`)?

```
1.16.4 v1.16.2-nightly-88
Go Version: go1.24.1
Git Ref: master
GitHash: c567110b2bc3a3d560fd6a04c04be538758a60e0
```

--- a/components/playground/instance/pd.go
+++ b/components/playground/instance/pd.go
@@ -15,9 +15,12 @@ package instance
 
 import (
 	"context"
+	"encoding/json"
 	"fmt"
+	"net/http"
 	"path/filepath"
 	"strings"
+	"time"
 
 	"github.com/pingcap/errors"
 	"github.com/pingcap/tiup/pkg/tidbver"
@@ -235,4 +238,41 @@ func (inst *PDInstance) LogFile() string {
 // Addr return the listen address of PD
 func (inst *PDInstance) Addr() string {
 	return utils.JoinHostPort(AdvertiseHost(inst.Host), inst.StatusPort)
+}
+
+// Ready returns nil when PD is ready to serve.
+func (inst *PDInstance) Ready(ctx context.Context) error {
+	url := fmt.Sprintf("http://%s/pd/api/v1/members", inst.Addr())
+
+	var r struct {
+		Header struct {
+			ClusterID uint64 `json:"cluster_id"`
+		} `json:"header"`
+	}
+
+	ready := func() bool {
+		resp, err := http.Get(url)
+		if err != nil {
+			return false
+		}
+		defer resp.Body.Close()
+		if resp.StatusCode == 200 {
+			err = json.NewDecoder(resp.Body).Decode(&r)
+			return err == nil && r.Header.ClusterID != 0
+		}
+		return false
+	}
+
+	for {
+		if ready() {
+			return nil
+		}
+
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case <-time.After(time.Second):
+			// retry
+		}
+	}
 }

--- a/components/playground/playground.go
+++ b/components/playground/playground.go
@@ -1285,6 +1285,7 @@ func (p *Playground) bootCluster(ctx context.Context, env *environment.Environme
 
 	anyPumpReady := false
 	allDMMasterReady := false
+	allPDReady := false
 	// Start all instance except tiflash.
 	err := p.WalkInstances(func(cid string, ins instance.Instance) error {
 		if cid == spec.ComponentTiFlash {
@@ -1294,6 +1295,19 @@ func (p *Playground) bootCluster(ctx context.Context, env *environment.Environme
 		if cid == spec.ComponentDMWorker && !allDMMasterReady {
 			p.waitAllDMMasterUp()
 			allDMMasterReady = true
+		}
+
+		// wait all PD ready before starting TiDB to avoid cluster ID mismatch
+		if cid == spec.ComponentTiDB && !allPDReady {
+			for _, pd := range p.pds {
+				pdCtx, cancel := context.WithTimeout(ctx, time.Second*120)
+				err := pd.Ready(pdCtx)
+				cancel()
+				if err != nil {
+					return errors.Annotatef(err, "failed to wait PD %s to be ready", pd.Addr())
+				}
+			}
+			allPDReady = true
 		}
 
 		err := p.startInstance(ctx, ins)

